src/Microsoft.AspNet.OData.Shared/TypelessPatchMethodHandler.cs

```diff
@@ -0,0 +1,196 @@
+        /// </summary>
+        /// <param name="keyValues">Key value pair for the entity keys</param>        
+        /// <param name="originalObject">Object to return</param>
+        /// <param name="errorMessage">Any error message in case of an exception</param>
+        /// <returns></returns>
+        public abstract PatchStatus TryGet(IDictionary<string, object> keyValues, out EdmStructuredObject originalObject, out string errorMessage);
+
+        /// <summary>
+        ///  TryDelete delegate to which the pointer to TryDelete Method can be assigned to, which will delete the object based on keyvalue pairs
+        /// </summary>
+        /// <param name="keyValues"></param>
+        /// <param name="errorMessage"></param>
+        /// <returns></returns>
+        public abstract PatchStatus TryDelete(IDictionary<string, object> keyValues, out string errorMessage);
+
+        /// <summary>
+        /// Get the PatchHandler for the nested type
+        /// </summary>
+        /// <param name="parent">Parent instance.</param>
+        /// <param name="navigationPropertyName">The name of the navigation property for the handler</param>
+        /// <returns></returns>
+        public abstract TypelessPatchMethodHandler GetNestedPatchHandler(EdmStructuredObject parent, string navigationPropertyName);
+    }
+
+       
+    internal class DefaultTypelessPatchHandler : TypelessPatchMethodHandler
+    {
+        IEdmEntityType entityType;
+        ICollection<EdmStructuredObject> originalList;
+
+        public DefaultTypelessPatchHandler(ICollection<EdmStructuredObject> originalList, IEdmEntityType entityType)
+        {
+            this.entityType = entityType;
+            this.originalList = originalList?? new List<EdmStructuredObject>();
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override PatchStatus TryGet(IDictionary<string, object> keyValues, out EdmStructuredObject originalObject, out string errorMessage)
+        {
+            PatchStatus status = PatchStatus.Success;
+            errorMessage = string.Empty;
+            originalObject = default(EdmStructuredObject);
+
+            try
+            {
+                originalObject = GetFilteredItem(keyValues);
+
+                if (originalObject == null)
+                {
+                    status = PatchStatus.NotFound;
+                }
+            }
+            catch (Exception ex)
+            {
+                status = PatchStatus.Failure;
+                errorMessage = ex.Message;
+            }
+
+            return status;
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override PatchStatus TryCreate(IEdmChangedObject changedObject, out EdmStructuredObject createdObject, out string errorMessage)
```

Comment:
>TryCreate [](start = 36, length = 9)

Remind me why we thought that TryCreate shouldn't set the values from changedObject?  is it because we had to set related properties as well, so we left it to Patch to handle setting all of the properties?  It seems like TryCreate should set the (non-entity) properties specified in changedObject, and then we should separately set the related entity-valued (navigation) properties.

Reply:
Patch will also patch (set all structural properties right?